internal/yggdrasil/configuration-assembler.go

```diff
@@ -74,24 +76,32 @@ func (a *configurationAssembler) getDeviceConfiguration(ctx context.Context, edg
 		Secrets:       secretList,
 	}
 
-	dc.Configuration.Heartbeat = getHeartbeatConfiguration(edgeDevice)
-
-	if edgeDevice.Spec.OsInformation != nil {
-		dc.Configuration.Os = (*models.OsInformation)(edgeDevice.Spec.OsInformation)
+	var deviceGroup *v1alpha1.EdgeDeviceGroup
+	if deviceGroupName, ok := edgeDevice.Labels["flotta/configured-by"]; ok {
+		logger.V(1).Info("Device uses EdgeDeviceGroup", "edgeDeviceGroup", deviceGroupName)
+		var err error
+		deviceGroup, err = a.groupRepository.Read(ctx, deviceGroupName, edgeDevice.Namespace)
+		if err != nil {
+			logger.Info("Cannot load EdgeDeviceGroup", "edgeDeviceGroup", deviceGroupName)
+		}
 	}
 
-	err := a.setStorageConfiguration(ctx, edgeDevice, &dc)
+	dc.Configuration.Heartbeat = getHeartbeatConfiguration(edgeDevice, deviceGroup)
```

Comment:
Pls document the behavior in the user guide, so the precedence of the configuration is known and its granularity

Reply:
Done.